src/ufs.rs

```diff
@@ -62,16 +62,40 @@ impl Ufs {
 		if blkno < nd {
 			Ok(NonZeroU64::new(direct[blkno as usize] as u64))
 		} else if blkno < (nd + pbp) {
+			let x = blkno - nd;
+			let low = x % pbp;
+
 			let first = indirect[0] as u64;
-			let pos = first * fs + (blkno - nd) * su64;
-			let pos: u64 = self.file.decode_at(pos)?;
-			Ok(NonZeroU64::new(pos))
+			let pos = first * fs + low * su64;
+			let block: u64 = self.file.decode_at(pos)?;
```

Comment:
This is a land mine.  It will work when you're mounting a regular file.  However, if you try to mount a device file it will fail.  That's because devices generally can't be read except in whole blocks, at block-aligned offsets.  So you should plan to read the entire fragment into memory and decode it, and then lookup the daddr that you want.

Reply:
`BlockReader` should abstract it away. That's why I wrote it. It reads a whole fragment/block and then operates in memory on it.
On top of `BlockReader` there is `Decoder`, which is what `self.file` is.
I should have probably named `self.file` to something else.

Reply:
Ahh, I see.  I was forgetting about BlockReader.